Adapt templates Eclipse adaption

### Expected behavior

As a Eclipse CobiGen user, I want update and adapt Templates command from Eclipse to work properly. 

### Actual behavior

Update and adapt templates command do not work and the following error will be thrown:

```Could not download file from https://repository.sonatype.org/service/local/artifact/maven/redirect?r=central-proxy&g=com.devonfw.cobigen&a=templates-devon4j&v=LATEST&c=sources
Could not download file from https://repository.sonatype.org/service/local/artifact/maven/redirect?r=central-proxy&g=com.devonfw.cobigen&a=templates-devon4j&v=LATEST&c=sources
  com.devonfw.cobigen.api.exception.CobiGenRuntimeException: Could not download file from https://repository.sonatype.org/service/local/artifact/maven/redirect?r=central-proxy&g=com.devonfw.cobigen&a=templates-devon4j&v=LATEST&c=sources
  	at com.devonfw.cobigen.api.util.TemplatesJarUtil.downloadJar(TemplatesJarUtil.java:87)
  	at com.devonfw.cobigen.api.util.TemplatesJarUtil.downloadLatestDevon4jTemplates(TemplatesJarUtil.java:101)
  	at com.devonfw.cobigen.eclipse.common.tools.ResourcesPluginUtil.downloadJar(ResourcesPluginUtil.java:178)
  	at com.devonfw.cobigen.eclipse.workbenchcontrol.handler.AdaptTemplatesHandler.execute(AdaptTemplatesHandler.java:80)
  	at org.eclipse.ui.internal.handlers.HandlerProxy.execute(HandlerProxy.java:294)
  	at org.eclipse.ui.internal.handlers.E4HandlerProxy.execute(E4HandlerProxy.java:96)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
  	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:58)
  	at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:320)
  	at org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:254)
  	at org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:164)
  	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:156)
  	at org.eclipse.core.commands.Command.executeWithChecks(Command.java:497)
  	at org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:490)
  	at org.eclipse.e4.core.commands.internal.HandlerServiceImpl.executeHandler(HandlerServiceImpl.java:213)
  	at org.eclipse.e4.ui.workbench.renderers.swt.HandledContributionItem.executeItem(HandledContributionItem.java:438)
  	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.handleWidgetSelection(AbstractContributionItem.java:449)
  	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.lambda$2(AbstractContributionItem.java:475)
  	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
  	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4131)
  	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1055)
  	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:3944)
  	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3547)
  	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1173)
  	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
  	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1062)
  	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:156)
  	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:636)
  	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
  	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:563)
  	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:151)
  	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:155)
  	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
  	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:137)
  	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:107)
  	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:400)
  	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
  	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:659)
  	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:595)
  	at org.eclipse.equinox.launcher.Main.run(Main.java:1501)
  	at org.eclipse.equinox.launcher.Main.main(Main.java:1474)
  Caused by: java.nio.file.NoSuchFileException: C:\Projects\testNewDevon\conf\.cobigen\templates\templates-devon4j-2021.12.006-sources.jar
  	at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:85)
  	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
  	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)
  	at java.base/sun.nio.fs.WindowsFileSystemProvider.newByteChannel(WindowsFileSystemProvider.java:235)
  	at java.base/java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:478)
  	at java.base/java.nio.file.Files.newOutputStream(Files.java:220)
  	at java.base/java.nio.file.Files.copy(Files.java:3067)
  	at com.devonfw.cobigen.api.util.TemplatesJarUtil.downloadJar(TemplatesJarUtil.java:78)
  	... 47 more
```

### Steps to reproduce (bug) / Use Case of feature request (enhancement)

1. Right click in Eclipse, choose CobiGen -> Adapt Templates

### Related/Dependent Issues
#1509 

### Comments/Hints:


### Affected version:

* OS: Windows/Linux/Mac?
* Browser: Chrome/Firefox/Safari?

--- a/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/AdaptTemplatesTest.java
+++ b/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/AdaptTemplatesTest.java
@@ -80,7 +80,8 @@ public class AdaptTemplatesTest extends SystemTest {
     javaClassItem.select();
 
     // execute CobiGen
-    EclipseCobiGenUtils.processCobiGen(bot, javaClassItem, 25000, "CRUD devon4j Server>CRUD REST services");
+    EclipseCobiGenUtils.processCobiGenAndPostponeUpgrade(bot, javaClassItem, "CRUD devon4j Server>CRUD REST services");
+
     bot.waitUntil(new AllJobsAreFinished(), 10000);
     // increase timeout as the openAPI parser is slow on initialization
     EclipseCobiGenUtils.confirmSuccessfullGeneration(bot, 40000);

--- a/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/AdaptTemplatesTest.java
+++ b/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/AdaptTemplatesTest.java
@@ -6,6 +6,7 @@ import static org.assertj.core.api.Assertions.assertThat;
 import java.io.File;
 
 import org.apache.commons.io.FileUtils;
+import org.eclipse.core.resources.IFile;
 import org.eclipse.core.resources.IProject;
 import org.eclipse.core.resources.IResource;
 import org.eclipse.core.resources.ResourcesPlugin;
@@ -19,9 +20,9 @@ import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
 
-import com.devonfw.cobigen.api.constants.ConfigurationConstants;
 import com.devonfw.cobigen.eclipse.common.constants.external.ResourceConstants;
 import com.devonfw.cobigen.eclipse.test.common.SystemTest;
+import com.devonfw.cobigen.eclipse.test.common.swtbot.AllJobsAreFinished;
 import com.devonfw.cobigen.eclipse.test.common.utils.EclipseCobiGenUtils;
 import com.devonfw.cobigen.eclipse.test.common.utils.EclipseUtils;
 
@@ -31,71 +32,77 @@ import com.devonfw.cobigen.eclipse.test.common.utils.EclipseUtils;
  */
 public class AdaptTemplatesTest extends SystemTest {
 
-    /** Temporary files rule to create temporary folders or files */
-    @Rule
-    public TemporaryFolder tempFolder = new TemporaryFolder();
+  /** Temporary files rule to create temporary folders or files */
+  @Rule
+  public TemporaryFolder tempFolder = new TemporaryFolder();
 
-    /** Root path of the Test Resources */
-    private static final String resourcesRootPath = "src/main/resources/OpenAPITest/";
+  /** Root path of the Test Resources */
+  private static final String resourcesRootPath = "src/main/resources/OpenAPITest/";
 
-    /** Line separator, e.g. for windows '\r\n' */
-    public static final String LINE_SEPARATOR = java.lang.System.getProperty("line.separator");
+  /** Line separator, e.g. for windows '\r\n' */
+  public static final String LINE_SEPARATOR = java.lang.System.getProperty("line.separator");
 
-    /**
-     * Setup workbench appropriately for tests
-     *
-     * @throws Exception test fails
-     */
-    @BeforeClass
-    public static void setupClass() throws Exception {
+  /**
+   * Setup workbench appropriately for tests
+   *
+   * @throws Exception test fails
+   */
+  @BeforeClass
+  public static void setupClass() throws Exception {
 
-        EclipseUtils.cleanWorkspace(bot, true);
+    EclipseUtils.cleanWorkspace(bot, true);
+  }
 
-    }
+  /**
+   * Test for external projects (not in workspace) taken as input for generation
+   *
+   * @throws Exception test fails
+   */
+  public void testBasicOpenAPIGenerationWithAdaptTemplates() throws Exception {
 
-    /**
-     * Test generation with OpenAPI input after adapting templates
-     *
-     * @throws Exception test fails
-     */
-    public void testBasicOpenAPIGenerationWithAdaptTemplates() throws Exception {
+    // copy sample project to external location and import it into the workspace
+    String testProjName = "ExtTestProj";
+    IJavaProject project = this.tmpMavenProjectRule.createProject(testProjName);
+    FileUtils.copyFile(new File(resourcesRootPath + "input/adapt-templates.yml"),
+        project.getUnderlyingResource().getLocation().append("adapt-templates.yml").toFile());
+    project.getProject().refreshLocal(IResource.DEPTH_INFINITE, new NullProgressMonitor());
+    this.tmpMavenProjectRule.updateProject();
 
-        // copy sample project to external location and import it into the workspace
-        String testProjName = "ExtTestProj";
-        IJavaProject project = this.tmpMavenProjectRule.createProject(testProjName);
-        FileUtils.copyFile(new File(resourcesRootPath + "input/adapt-templates.yml"),
-                project.getUnderlyingResource().getLocation().append("adapt-templates.yml").toFile());
-        project.getProject().refreshLocal(IResource.DEPTH_INFINITE, new NullProgressMonitor());
-        this.tmpMavenProjectRule.updateProject();
+    EclipseCobiGenUtils.runAndCaptureUpdateTemplates(bot);
+    EclipseCobiGenUtils.runAndCaptureAdaptTemplates(bot);
+    EclipseUtils.updateMavenProject(bot, ResourceConstants.CONFIG_PROJECT_NAME);
 
-        EclipseCobiGenUtils.runAndCaptureUpdateTemplates(bot);
-        EclipseCobiGenUtils.runAndCaptureAdaptTemplates(bot);
-        EclipseUtils.updateMavenProject(bot, ResourceConstants.CONFIG_PROJECT_NAME);
+    EclipseUtils.openErrorsTreeInProblemsView(bot);
 
-        EclipseUtils.openErrorsTreeInProblemsView(bot);
+    // expand the new file in the package explorer
+    SWTBotView view = bot.viewById(JavaUI.ID_PACKAGES);
+    SWTBotTreeItem javaClassItem = view.bot().tree().expandNode(testProjName, "adapt-templates.yml");
+    javaClassItem.select();
 
-        // expand the new file in the package explorer
-        SWTBotView view = bot.viewById(JavaUI.ID_PACKAGES);
-        SWTBotTreeItem javaClassItem = view.bot().tree().expandNode(testProjName, "adapt-templates.yml");
-        javaClassItem.select();
+    // execute CobiGen
+    EclipseCobiGenUtils.processCobiGen(bot, javaClassItem, 25000, "CRUD devon4j Server>CRUD REST services");
+    bot.waitUntil(new AllJobsAreFinished(), 10000);
+    // increase timeout as the openAPI parser is slow on initialization
+    EclipseCobiGenUtils.confirmSuccessfullGeneration(bot, 40000);
 
-        IProject adaptedTemplatesProj = ResourcesPlugin.getWorkspace().getRoot()
-                .getProject(ConfigurationConstants.COBIGEN_TEMPLATES);
+    bot.waitUntil(new AllJobsAreFinished(), 10000);
+    IProject proj = ResourcesPlugin.getWorkspace().getRoot().getProject(testProjName);
+    IFile generationResult = proj.getFile(
+        "src/main/java/com/devonfw/test/sampledatamanagement/service/impl/rest/SampledatamanagementRestServiceImpl.java");
 
-        assertThat(adaptedTemplatesProj.exists()).isTrue();
-    }
+    assertThat(generationResult.exists()).isTrue();
+  }
 
-    /**
-     * Test of testBasicOpenAPIGenerationWithAdaptTemplates with custom COBIGEN_HOME
-     * environment variable
-     *
-     * @throws Exception test fails
-     */
-    @Test
-    public void testAdaptTemplates() throws Exception {
+  /**
+   * Test of testBasicOpenAPIGenerationWithAdaptTemplates with custom COBIGEN_HOME environment variable
+   *
+   * @throws Exception test fails
+   */
+  @Test
+  public void testAdaptTemplatesAndGenerate() throws Exception {
 
-        File tmpProject = this.tempFolder.newFolder("playground", "project", "templates");
-        withEnvironmentVariable(ConfigurationConstants.CONFIG_ENV_HOME, tmpProject.getParentFile().getAbsolutePath())
-                .execute(() -> testBasicOpenAPIGenerationWithAdaptTemplates());
-    }
+    File tmpProject = this.tempFolder.newFolder("playground", "project");
+    withEnvironmentVariable("COBIGEN_HOME", tmpProject.toPath().toString())
+        .execute(() -> testBasicOpenAPIGenerationWithAdaptTemplates());
+  }
 }

--- a/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/AdaptTemplatesTest.java
+++ b/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/AdaptTemplatesTest.java
@@ -20,6 +20,8 @@ import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
 
+import com.devonfw.cobigen.api.util.CobiGenPaths;
+import com.devonfw.cobigen.api.util.TemplatesJarUtil;
 import com.devonfw.cobigen.eclipse.common.constants.external.ResourceConstants;
 import com.devonfw.cobigen.eclipse.test.common.SystemTest;
 import com.devonfw.cobigen.eclipse.test.common.swtbot.AllJobsAreFinished;
@@ -68,7 +70,9 @@ public class AdaptTemplatesTest extends SystemTest {
     project.getProject().refreshLocal(IResource.DEPTH_INFINITE, new NullProgressMonitor());
     this.tmpMavenProjectRule.updateProject();
 
-    EclipseCobiGenUtils.runAndCaptureUpdateTemplates(bot);
+    File templatesDirectory = CobiGenPaths.getTemplatesFolderPath().toFile();
+    TemplatesJarUtil.downloadLatestDevon4jTemplates(true, templatesDirectory);
+    TemplatesJarUtil.downloadLatestDevon4jTemplates(false, templatesDirectory);
     EclipseCobiGenUtils.runAndCaptureAdaptTemplates(bot);
     EclipseUtils.updateMavenProject(bot, ResourceConstants.CONFIG_PROJECT_NAME);
 

--- a/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/AdaptTemplatesTest.java
+++ b/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/AdaptTemplatesTest.java
@@ -91,6 +91,7 @@ public class AdaptTemplatesTest extends SystemTest {
         "src/main/java/com/devonfw/test/sampledatamanagement/service/impl/rest/SampledatamanagementRestServiceImpl.java");
 
     assertThat(generationResult.exists()).isTrue();
+
   }
 
   /**

--- a/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/common/utils/EclipseCobiGenUtils.java
+++ b/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/common/utils/EclipseCobiGenUtils.java
@@ -3,6 +3,8 @@ package com.devonfw.cobigen.eclipse.test.common.utils;
 import static org.eclipse.swtbot.swt.finder.waits.Conditions.shellIsActive;
 import static org.eclipse.swtbot.swt.finder.waits.Conditions.widgetIsEnabled;
 
+import java.util.concurrent.TimeoutException;
+
 import org.eclipse.core.resources.IncrementalProjectBuilder;
 import org.eclipse.core.resources.ResourcesPlugin;
 import org.eclipse.core.runtime.CoreException;
@@ -16,8 +18,6 @@ import org.eclipse.swtbot.swt.finder.SWTBot;
 import org.eclipse.swtbot.swt.finder.widgets.SWTBotButton;
 import org.eclipse.swtbot.swt.finder.widgets.SWTBotShell;
 import org.eclipse.swtbot.swt.finder.widgets.SWTBotTreeItem;
-import org.eclipse.swtbot.swt.finder.widgets.TimeoutException;
-import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import com.devonfw.cobigen.eclipse.common.constants.external.CobiGenDialogConstants;
@@ -393,10 +393,6 @@ public class EclipseCobiGenUtils {
     takeScreenshot(bot, "Adapt Templates Warning!");
     SWTBotShell warningDialog = bot.shell("Warning!");
     warningDialog.bot().button("Ok").click();
-
-    takeScreenshot(bot, "Create new POM!");
-    SWTBotShell finishDialog = bot.shell("Create new POM");
-    finishDialog.bot().button("Finish").click();
 
     SWTBotShell informationDialog = bot.shell("Information");
     bot.waitUntil(new AnyShellIsActive("Information"), DEFAULT_TIMEOUT);

--- a/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/common/utils/EclipseCobiGenUtils.java
+++ b/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/common/utils/EclipseCobiGenUtils.java
@@ -3,8 +3,6 @@ package com.devonfw.cobigen.eclipse.test.common.utils;
 import static org.eclipse.swtbot.swt.finder.waits.Conditions.shellIsActive;
 import static org.eclipse.swtbot.swt.finder.waits.Conditions.widgetIsEnabled;
 
-import java.util.concurrent.TimeoutException;
-
 import org.eclipse.core.resources.IncrementalProjectBuilder;
 import org.eclipse.core.resources.ResourcesPlugin;
 import org.eclipse.core.runtime.CoreException;
@@ -18,6 +16,8 @@ import org.eclipse.swtbot.swt.finder.SWTBot;
 import org.eclipse.swtbot.swt.finder.widgets.SWTBotButton;
 import org.eclipse.swtbot.swt.finder.widgets.SWTBotShell;
 import org.eclipse.swtbot.swt.finder.widgets.SWTBotTreeItem;
+import org.eclipse.swtbot.swt.finder.widgets.TimeoutException;
+import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import com.devonfw.cobigen.eclipse.common.constants.external.CobiGenDialogConstants;

--- a/cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/common/tools/ResourcesPluginUtil.java
+++ b/cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/common/tools/ResourcesPluginUtil.java
@@ -13,9 +13,6 @@ import org.eclipse.core.resources.IProject;
 import org.eclipse.core.resources.IResource;
 import org.eclipse.core.resources.ResourcesPlugin;
 import org.eclipse.core.runtime.CoreException;
-import com.devonfw.cobigen.eclipse.common.constants.external.ResourceConstants;
-import com.devonfw.cobigen.eclipse.common.exceptions.GeneratorProjectNotExistentException;
-import com.devonfw.cobigen.eclipse.updatetemplates.UpdateTemplatesDialog;
 import org.eclipse.core.runtime.IPath;
 import org.eclipse.core.runtime.NullProgressMonitor;
 import org.eclipse.jface.dialogs.MessageDialog;
@@ -24,12 +21,13 @@ import org.eclipse.swt.widgets.Display;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-
-
 import com.devonfw.cobigen.api.TemplateAdapter;
 import com.devonfw.cobigen.api.constants.ConfigurationConstants;
 import com.devonfw.cobigen.api.util.CobiGenPaths;
 import com.devonfw.cobigen.api.util.TemplatesJarUtil;
+import com.devonfw.cobigen.eclipse.common.constants.external.ResourceConstants;
+import com.devonfw.cobigen.eclipse.common.exceptions.GeneratorProjectNotExistentException;
+import com.devonfw.cobigen.eclipse.updatetemplates.UpdateTemplatesDialog;
 import com.devonfw.cobigen.impl.CobiGenFactory;
 import com.devonfw.cobigen.impl.adapter.TemplateAdapterImpl;
 
@@ -261,7 +259,7 @@ public class ResourcesPluginUtil {
     }
 
     try {
-      TemplateAdapter templateAdapter = new TemplateAdapterImpl(cobigenFolderPath);
+      TemplateAdapter templateAdapter = new TemplateAdapterImpl(null);
       templateAdapter.adaptMonolithicTemplates(cobigenFolderPath.resolve(ConfigurationConstants.COBIGEN_TEMPLATES),
           false);
     } catch (Exception e) {
@@ -297,7 +295,6 @@ public class ResourcesPluginUtil {
   public static void startTemplatesUpgrader(Path configurationPath) {
 
     setTemplateSetPathAfterUpgrade(CobiGenFactory.startTemplatesUpgrader(configurationPath));
-
 
     // FIXME after the upgrade adapt the new template-set folder. should be done in #1587
   }

--- a/cobigen/cobigen-core-api/src/main/java/com/devonfw/cobigen/api/util/CobiGenPaths.java
+++ b/cobigen/cobigen-core-api/src/main/java/com/devonfw/cobigen/api/util/CobiGenPaths.java
@@ -75,7 +75,9 @@ public class CobiGenPaths {
   public static Path getTemplatesFolderPath(Path home) {
 
     Path templatesPath = home.resolve(ConfigurationConstants.TEMPLATES_FOLDER);
-    return templatesPath;
+
+    return createFolder(templatesPath);
+
   }
 
   /**

